base/strings/util.jl

```diff
@@ -45,18 +39,8 @@ true
 ```
 """
 function endswith(a::AbstractString, b::AbstractString)
-    i = lastindex(a)
-    j = lastindex(b)
-    a1 = start(a)
-    b1 = start(b)
-    while a1 <= i && b1 <= j
-        c = a[i]
-        d = b[j]
-        (c != d) && (return false)
-        i = prevind(a,i)
-        j = prevind(b,j)
-    end
-    j < b1
+    a, b = Iterators.Stateful.(Iterators.reverse.((a,b)))
```

Comment:
The goal isn't to write exceptionally clever code - this doesn't need to use `broadcast`, it can just be explicit

Reply:
> The goal isn't to write exceptionally clever code

but if it were, I feel like I'd be winning with this PR. I'll change it back.